### ISSUE TITLE
ClientSecret Auth Type

### ISSUE THREAD

**Issue by ryanlraines** [seanmcne/Microsoft.Xrm.Data.PowerShell#470]
Attempting to authenticate against AAD with ClientID & ClientSecret. 

Steps to reproduce:
1. AAD App Registration ID and matching secret
2. App Registration group permissions granted to match CRM Org security group
3. App Registration Redirect URL is configured to be same as the the CRM Org url 
3. Run this `Connect-CrmOnline -ClientSecret "FOO" -OAuthClientId "BAR" -ServerUrl https://foobar.crm.microsoftdynamics.us/"`

Results:
```text
IsReady                             : False
IsBatchOperationsAvailable          : False
MaxRetryCount                       : 10
RetryPauseTime                      : 00:00:05
Authority                           : https://login.microsoftonline.us/8675309/oauth2/authorize/
ActiveAuthenticationType            : ClientSecret
OAuthUserId                         :
TenantId                            : 00000000-0000-0000-0000-000000000000
EnvironmentId                       :
ConnectedOrgId                      : 00000000-0000-0000-0000-000000000000
CrmConnectOrgUriActual              : https://foobar.crm.microsoftdynamics.us/XRMServices/2011/Organization.svc/web?SDKClientVersion=9.0.45.2156
ConnectedOrgFriendlyName            :
ConnectedOrgUniqueName              :
ConnectedOrgVersion                 :
SdkVersionProperty                  : 9.0.45.2156
CallerId                            : 00000000-0000-0000-0000-000000000000
CallerAADObjectId                   :
DisableCrossThreadSafeties          : False
SessionTrackingId                   :
ForceServerMetadataCacheConsistency : False
LastCrmError                        : Invalid Login Information : The HTTP request was forbidden with client authentication scheme 'Anonymous'.
                                      The HTTP request was forbidden with client authentication scheme 'Anonymous'. => The remote server returned an error: (403) Forbidden.Unable to Login to Dynamics CRM
                                      Unable to Login to Dynamics CRMOrganizationWebProxyClient is null
                                      OrganizationWebProxyClient is null
```

**Comment by seanmcne** [comment]
Try adding -verbose this will show you the connection string to help diagnose along with other details. Also, check your AAD sign in logs for any indication of the issue. Finally, please make sure "bar" is a GUID matching  your client app ID, ensure the app user has been created as an app user in the dataverse environment, ensure the app user has a role in the dataverse environment, and verify the delegated user auth is enabled in AAD for the app user. Thx!

**Comment by ryanlraines** [comment]
@seanmcne  Thanks for the reply. 

I was able to resolve the issue, and your comments are aligned with how I resolved it. 

The missing step was, as you suggested, adding the AAD Client as an Application User in CRM/Dataverse. Also the user must be granted the appropriate role. 

Ref Doc: https://docs.microsoft.com/en-us/power-platform/admin/create-users-assign-online-security-roles#create-an-application-user
